### DISCUSSION TITLE
Simplified Download Options ⬇️

### DISCUSSION THREAD

**Discussion by e-Feitosa** [sircharlo/meeting-media-manager#3359] [Ideas] [answered]
### Is your idea related to a problem?

 It's not a problem, but I see many people having difficulty downloading the correct version among so many options. Therefore, I would like to suggest the following for a better understanding of the available versions for each operating system:

### Describe your idea

To simplify distribution and reduce the number of versions, here are some suggestions:

## 1. **Distribution by Architecture:** ##
- **Windows**: Keep only one version for 32-bit and one for 64-bit. 
- **Linux**: Keep only the AppImage file, which is compatible with most Linux distributions.
- **macOS**: Keep only one version for ARM and one for Intel, eliminating compressed files if they are not strictly necessary.

## 2. **Verification Files (Blockmap):** ##
- Consider including the verification files (blockmap) within the installers themselves, if possible, to reduce the number of separate files.

## 3. **Configuration Files**: ##
- Consolidate the configuration files (YML) into a single generic file if the differences between them are minimal.

## 4. **Source Code:** ##
- Keep only one compressed version of the source code, preferably in the most widely used format (ZIP or TAR.GZ).

### Describe alternatives you've considered

Example for File Distribution

## **Windows** ##
- meeting-media-manager-32bits.exe - Installer for 32-bit Windows systems.
- meeting-media-manager-64bits.exe - Installer for 64-bit Windows systems.

## **Linux** ##
- meeting-media-manager.AppImage - Executable file for Linux systems.

## **macOS** ##
- meeting-media-manager-arm64.dmg - Installer for macOS with ARM processors.
- meeting-media-manager-x64.dmg - Installer for macOS with Intel processors.

## **Generic** ##
- latest.yml - Generic configuration file.

## **Source Code** ##
- Source code (zip) - Project source code in ZIP format.

### Additional context

The option below keeps the current list from the site, but with a brief explanation of each item. 

**Windows 64-bit**
- meeting-media-manager-24.11.3.exe (184 MB) - Complete installer for 64-bit Windows systems.
- meeting-media-manager-24.11.3-x64.exe (95.1 MB) - Installer for 64-bit Windows systems.
- meeting-media-manager-24.11.3-x64.exe.blockmap (103 KB) - Integrity verification file for the 64-bit installer.
- meeting-media-manager-24.11.3.exe.blockmap (195 KB) - Integrity verification file for the complete 64-bit installer.

**Windows 32-bit**
- meeting-media-manager-24.11.3-ia32.exe (89.6 MB) - Installer for 32-bit Windows systems.
- meeting-media-manager-24.11.3-ia32.exe.blockmap (97.6 KB) - Integrity verification file for the 32-bit installer.

**Linux**
- meeting-media-manager-24.11.3-x86_64.AppImage (129 MB) - Executable file for Linux systems.
- latest-linux.yml (414 Bytes) - Configuration file for installation on Linux.

**macOS Arm64**
- meeting-media-manager-24.11.3-arm64.dmg (120 MB) - Installer for macOS systems with ARM processors.
- meeting-media-manager-24.11.3-arm64.zip (115 MB) - Compressed file for macOS systems with ARM processors.
- meeting-media-manager-24.11.3-arm64.dmg.blockmap (130 KB) - Integrity verification file for the ARM installer.
- meeting-media-manager-24.11.3-arm64.zip.blockmap (124 KB) - Integrity verification file for the ARM compressed file.

**macOS Intel x64**
- meeting-media-manager-24.11.3-x64.dmg (125 MB) - Installer for macOS systems with Intel processors.
- meeting-media-manager-24.11.3-x64.zip (120 MB) - Compressed file for macOS systems with Intel processors.
- meeting-media-manager-24.11.3-x64.dmg.blockmap (134 KB) - Integrity verification file for the Intel installer.
- meeting-media-manager-24.11.3-x64.zip.blockmap (130 KB) - Integrity verification file for the Intel compressed file.

**Generic**
- latest.yml (700 Bytes) - Generic configuration file.

**Source Code**
- Source code (zip) - Project source code in ZIP format.
- Source code (tar.gz) - Project source code in TAR.GZ format.

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing issues to make sure my idea is not a duplicate.
- [X] I confirm that I have searched through the existing discussions to make sure my idea is not a duplicate.

**Comment by ux-git** [comment]
@sircharlo What about adding the Download button to the website with auto-detection of the OS and suggesting to downloading the correct version? With other operating systems still available on other tabs.

Then the choice will shrink down to the x32 vs x64 for Windows or ARM vs Intel for macOS.

The links can lead to to downloadable GitHub build, to download right away, but technically still from the GitHub. I'm just not sure whether it's possible to easily implement using the latest available version for download automatically, instead of manually updating the app all the time.

_________

This way less tech savvy friends will find the app on the website easy to download _(for many people it's even hard to navigate on GitHub to a point where it's not obviously that the app downloads are on the Releases page)_, and for those who want to dive deeper – GitHub is the way to go.

**Reply by mtdvlpr** [reply]
Looks like it's quite easy to implement: https://docs.github.com/en/repositories/releasing-projects-on-github/linking-to-releases#linking-to-the-latest-release

**Comment by mtdvlpr** [comment, marked as answer]
I've added direct download links to the documentation. They should automatically download the latest version from GitHub.

https://sircharlo.github.io/meeting-media-manager/using-at-a-kingdom-hall#download-and-install

**Reply by ux-git** [reply]
Nice. A big DOWNLOAD button on the home page would also be nice to have.

**Reply by mtdvlpr** [reply]
That won't be possible unfortunately. The homepage is made from a template using configuration values, so there's no way to add custom behavior to the action buttons.

**Reply by ux-git** [reply]
A simple link to the downloads paragraph would be a good compromise. Just to simplify the navigation.

**Reply by mtdvlpr** [reply]
That we could do! Instead of linking to the latest github release, we link to the download section.

**Reply by mtdvlpr** [reply]
Done :)
